examples/context-propagation/kotlin/src/main/kotlin/example/armeria/contextpropagation/manual/MainService.kt

```diff
@@ -0,0 +1,104 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package example.armeria.contextpropagation.manual
+
+import com.linecorp.armeria.client.WebClient
+import com.linecorp.armeria.common.AggregatedHttpResponse
+import com.linecorp.armeria.common.HttpRequest
+import com.linecorp.armeria.common.HttpResponse
+import com.linecorp.armeria.server.HttpService
+import com.linecorp.armeria.server.ServiceRequestContext
+import com.google.common.base.Splitter
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.Iterables
+import com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly
+import java.time.Duration
+import java.util.stream.Collectors
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.future.asDeferred
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.future.future
+import kotlinx.coroutines.withContext
+
+class MainService(private val backendClient: WebClient) : HttpService {
+    override fun serve(ctx: ServiceRequestContext, req: HttpRequest): HttpResponse {
+        val ctxExecutor = ctx.contextAwareExecutor()
+        val response = GlobalScope.future(ctxExecutor.asCoroutineDispatcher()) {
+            val numsFromRequest = async { fetchFromRequest(ctx, req) }
+            val numsFromDb = async { fetchFromFakeDb(ctx) }
+            val nums = awaitAll(numsFromRequest, numsFromDb).flatten()
+
+            val backendResponses =
+                awaitAll(
+                    *nums.map { num ->
+                        // The context is mounted in a thread-local, meaning it is available to all logic such
+                        // as tracing.
+                        require(ServiceRequestContext.current() === ctx)
+                        require(ctx.eventLoop().inEventLoop())
+                        backendClient.get("/square/$num").aggregate().asDeferred()
+                    }.toTypedArray()
+                ).toList()
+            // The context is mounted in a thread-local, meaning it is available to all logic such as tracing.
+            require(ServiceRequestContext.current() === ctx)
+            require(ctx.eventLoop().inEventLoop())
+            HttpResponse.of(
+                backendResponses.stream()
+                    .map(AggregatedHttpResponse::contentUtf8)
+                    .collect(Collectors.joining("\n"))
+            )
+        }
+        return HttpResponse.from(response)
+    }
+
+    private suspend fun fetchFromRequest(ctx: ServiceRequestContext, req: HttpRequest): List<Long> {
+        // The context is mounted in a thread-local, meaning it is available to all logic such as tracing.
+        require(ServiceRequestContext.current() === ctx)
+        require(ctx.eventLoop().inEventLoop())
+        val aggregatedHttpRequest = req.aggregate().await()
```

Comment:
Can we check context / event loop after await too? I guess that's the equivalent of a callback in future style and it would be incorrect if the context was wrong here and we needed to make another backend call.